packages/sg721/src/lib.rs

```diff
@@ -92,31 +95,26 @@ pub struct UpdateCollectionInfoMsg<T> {
 }
 
 #[cw_serde]
-pub struct RoyaltyInfo {
-    pub payment_address: Addr,
+pub struct RoyaltyInfo<T: AddressLike> {
```

Comment:
This struct is also used by the marketplace, not sure what happens to serde when there is unexpected new fields.

Reply:
needs Option, fixed.